src/core/database/query.helpers.ts

```diff
@@ -103,77 +103,27 @@ export function matchRequestingUser(
   ]);
 }
 
-//DELETE service
-export const setBaseNodeLabelsAndIdDeleted = (
-  query: Query,
-  baseNodeLabels: string[]
-) => {
-  //set labels as Deleted
-  baseNodeLabels.forEach((label) => {
-    query.call(setLabelDeleted, 'node', label).with('distinct(node) as node');
-  });
-
-  //set id as deleted_id
-
-  query
-    .with('*, node.id as nodeId')
-    .set({
-      variables: {
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        'node.deleted_id': 'nodeId',
-      },
-    })
-    .removeProperties({
-      node: 'id',
-    })
-    .with('distinct(node) as node');
-};
-
-export type UniqueProperties<BaseNode extends Resource> = Partial<
-  Record<keyof BaseNode, string[]>
->;
-
-export function setPropLabelsAndValuesDeleted<BaseNode extends Resource>(
-  query: Query,
-  uniqueProperties: UniqueProperties<BaseNode>
-) {
-  entries(uniqueProperties).forEach(([property, labels], i) => {
-    const currentPropertyNodeAlias = `propertyNode${i}`;
-    // Match the baseNode out to the propertyNode
-    query.optionalMatch([
-      node('node'),
-      relation('out', '', property),
-      node(currentPropertyNodeAlias, 'Property'),
-    ]);
-
-    // Reset all the labels on the propertyNode with the Delete_ prefix
-    labels?.forEach((label) => {
-      query.call(setLabelDeleted, currentPropertyNodeAlias, label);
-    });
-
-    // Reset all the values on the propertyNode to deleted_value
-    query
-      .with(`*, ${currentPropertyNodeAlias}.value as propVal${i}`)
-      .set({
-        variables: {
-          [`${currentPropertyNodeAlias}.deleted_value`]: `propVal${i}`,
-        },
-      })
-      .removeProperties({
-        [currentPropertyNodeAlias]: 'value',
-      })
-      .with('*');
-  });
-}
-
-const setLabelDeleted = (query: Query, nodeAlas: string, label: string) => {
-  query
-    .set({
-      labels: {
-        [nodeAlas]: `Deleted_${label}`,
-      },
-    })
-    .removeLabels({
-      [nodeAlas]: label,
-    });
+/**
+ * This will set all relationships given to active false
+ * and add deleted prefix to its labels.
+ */
+export const deleteProperties = <Resource extends ResourceShape<any>>(
+  _resource: Resource,
```

Comment:
This looks great. Just tested it out. I'd say this is ready to go. Not sure why the checks aren't going through.